src/text-field/text-field.svelte

```diff
@@ -24,16 +24,17 @@
     console.error('Labels are only available for outlined text fields');
   }
 
-  let internalValue = null;
   export let value = null;
-  $: value = (
-    $$restProps.type === 'number' && internalValue != null ? +internalValue : internalValue
-  );
   export let events = null;
 
   let inputElement;
   onMount(() => autofocus && inputElement.focus());
 
+  function handleInput(event) {
+    value = $$restProps.type === 'number' ? +event.target.value : event.target.value;
```

Comment:
I wanted to suggest using `parseInt` instead of the unary plus because it seem more fitting, depending on the first 2 answers on [this SO post](https://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which) (especially the summary table in the 2nd answer), but then I decided to check the internal svelte implementation for `<input type="number">`, and turns out that they just do like this:
```js
function to_number(value) {
    return value === '' ? undefined : +value;
}
```
(It is worth mentioning that this can be imported using `import { to_number } from 'svelte/internal';`)
What do you think about this?

Reply:
Not keen on importing anything from `svelte/internal`. As for `parseInt`, yes, it would work better here.